### ISSUE TITLE
Parsing error when using different types of declarations

### ISSUE THREAD

**Issue by rbokel** [BinaryMuse/toml-node#40]
Not sure, if this is expected or not. 

The example below defines an array of objects (named top), that have an array of objects named (inner). 
When the array inner is empty, it is declared as inner = []. 
When it has content, it is declared as [[top.inner]] with some properties after 

The parser fails with 

`Cannot redefine existing key 'top.inner'.`

```
[[top]]  
id = 'top-1'  
inner = []  
[[top]]  
id = 'top-2'  
[[top.inner]]  
id = "inner-1"  
[[top.inner]]  
id = "inner-2"
```  

Expected structure 
```
top = [
  {id: 'top-1', inner: []},
  {id: 'top-2', inner: [{id: 'inner-1}, {id: 'inner-2'}]},
]
```

**Comment by felix9** [comment]
yeah, that's a bug. I have a fix, and I'll send a pull request in a bit

**Comment by BinaryMuse** [comment]
Thanks to @felix9, this is fixed in `v2.3.3`.
